internal/server/redirect_test.go

```diff
@@ -24,12 +23,10 @@ func TestVerifyAndRedirect_Works(t *testing.T) {
 	assert.Assert(t, len(user.VerificationToken) > 0, "verification token must be set")
 	assert.Assert(t, !user.Verified)
 
-	redirectURL := "https://example.com/hello"
+	redirectURL := "/hello"
 	url := wrapLinkWithVerification(redirectURL, "example.com", user.VerificationToken)
 
-	req, err := http.NewRequestWithContext(context.Background(), http.MethodGet, url, nil)
-	assert.NilError(t, err)
-
+	req := httptest.NewRequest(http.MethodGet, url, nil)
```

Comment:
This change is a bit surprising. It seems like `wrapLinkWithVerification` is expecting a full url, so passing in just a path doesn't seem correct.

`redirectURL` is used in a comparison below, I guess that's why the value changed?  

We always pass `wrapLinkWithVerification` a full URL in our production code paths, so we should probably do that here as well.

Reply:
`httptest.NewRequest` uses `example.com` as request host if the URL is a relative path which is why this has less impact than expected